src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -311,7 +339,72 @@ private RecurringDisbursementVoucherPDPStatus buildRecurringDisbursementVoucherP
         }
         return paymentDetails;
     }
- 
+    
+    private void noteChangeOnRecurringDV(RecurringDisbursementVoucherDocument recurringDV, String noteText, Set<String> setOfStrings) {
+        if (setOfStrings != null && !setOfStrings.isEmpty()) {
+            Note paymentGroupNote = buildNoteBase();
+            paymentGroupNote.setNoteText(noteText + StringUtils.join(setOfStrings, ", "));;
+            recurringDV.addNote(paymentGroupNote);
+            getBusinessObjectService().save(recurringDV);
+        }
+    }
+    
+    private Note buildNoteBase() {
+        Note noteBase = new Note();
+        noteBase.setAuthorUniversalIdentifier(GlobalVariables.getUserSession().getPerson().getPrincipalId());
+        noteBase.setNotePostedTimestampToCurrent();
+        return noteBase;
+    }
+    
+    @Override
+    public boolean isPaymentCancelable(Person user, String paymentDetailStatus) {
+        return paymentDetailStatus.equalsIgnoreCase(PdpConstants.PaymentStatusCodes.OPEN) 
+                && getRecurringDisbursementVoucherPaymentMaintenanceService().hasCancelPermission(user);
```

Comment:
If you rewrite the first part as "PdpConstants.PaymentStatusCode.OPEN.equalsIgnoreCase(paymentDetailStatus)", then you can remove the extra null check in the cancelPDPPayments() method above.